src/k8s/cmd/k8s/k8s_bootstrap.go

```diff
@@ -125,16 +143,13 @@ func newBootstrapCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
 }
 
 func getConfigFromYaml(filePath string) (apiv1.BootstrapConfig, error) {
-	config := apiv1.BootstrapConfig{}
-	config.SetDefaults()
-
-	yamlContent, err := os.ReadFile(filePath)
+	b, err := os.ReadFile(filePath)
 	if err != nil {
-		return config, fmt.Errorf("failed to read YAML config file: %w", err)
+		return apiv1.BootstrapConfig{}, fmt.Errorf("failed to read file: %w", err)
 	}
 
-	err = yaml.Unmarshal(yamlContent, &config)
-	if err != nil {
+	var config apiv1.BootstrapConfig
+	if err := yaml.UnmarshalStrict(b, &config); err != nil {
 		return config, fmt.Errorf("failed to parse YAML config file: %w", err)
```

Comment:
same why does this return a valid default bootstrap config and not `nil`

Reply:
see above